yjit/src/codegen.rs

```diff
@@ -816,56 +817,112 @@ gen_single_block(blockid_t blockid, const ctx_t *start_ctx, rb_execution_context
 
 
 
-fn gen_nop(jit: &JITState, ctx: &mut Context, cb: &CodeBlock) -> CodegenStatus
+fn gen_nop(jit: &JITState, ctx: &mut Context, cb: &mut CodeBlock) -> CodegenStatus
 {
     // Do nothing
-    return KeepCompiling;
+    KeepCompiling
 }
 
-fn gen_pop(jit: &JITState, ctx: &mut Context, cb: &CodeBlock) -> CodegenStatus
+fn gen_pop(jit: &JITState, ctx: &mut Context, cb: &mut CodeBlock) -> CodegenStatus
 {
     // Decrement SP
     ctx.stack_pop(1);
-    return KeepCompiling;
+    KeepCompiling
+}
+
+fn gen_dup(jit: &JITState, ctx: &mut Context, cb: &mut CodeBlock) -> CodegenStatus
+{
+    let dup_val = ctx.stack_pop(1);
+    let (mapping, tmp_type) = ctx.get_opnd_mapping(StackOpnd(0));
+
+    let loc0 = ctx.stack_push_mapping((mapping, tmp_type));
+    //mov(cb, REG0, dup_val);  // Huh. Mov() isn't implemented. Why did I think it was?
+    //mov(cb, loc0, REG0);
```

Comment:
If you comment out crucial bits of code, you should write a `todo!()` here so that execution will result in a panic at runtime. Otherwise there's a risk of silent failure.

Reply:
Okay. But then none of the tests will pass. Should I not be checking in the tests?